docs/future_plans.md

```diff
@@ -0,0 +1,69 @@
+**Last updated:** January 2021.
+
+**Summary:** We'll be redesigning how Stardoc works, and deprioritizing feature
+requests and minor bugs until that work is complete (targeting 2021 Q2).
+
+## Technical motivation
+
+Stardoc is currently the recommended tool for generating documentation of
+Starlark rules. It [replaces](skydoc_deprecation.md) *Sky*doc, the previous
+tool, which worked by evaluating .bzl files in a Python interpreter, using
+fake versions of functions from Bazel's Build Language.
+
+Mocking is an inherently problematic approach for two reasons:
+
+1. It creates a maintenance burden for the tool maintainer (us). We have to
+ensure that the mocked definitions stay up-to-date as Bazel changes. These
+include not just `rule()` and `provider()`, but also a number of other symbols
+that don't directly affect documentation but still require stubs.
+
+2. It puts a constraint on the user: All of their documented .bzl files, as
+well as all of the .bzl dependencies they transitively load, must be
+compatible with the mock evaluation. This means users must be vigilant about
+writing Starlark code that lies in the intersection of what is understood as
+valid by Bazel and by Stardoc.
+
+The Python-based Skydoc experienced an extreme version of this problem because
+it didn't even treat .bzl files as being written in the Starlark language.
+However, the Java-based Stardoc still uses mocking -- not of the Starlark
+language, but of Bazel's Build Language functions.
+
+In addition, Stardoc's mocking approach tightly integrates it with Bazel.
+Indeed much of its source code lives inside the bazelbuild/bazel repository.
+This makes refactoring and evolving the Bazel source code more difficult.
+
+While the Starlark language has a specification and several implementations,
+the Build Language is more complicated and has only one accurate
+implementation: Bazel. Any tooling that operates on BUILD and .bzl files must
+carefully consider whether it is feasible to ask Bazel for the authoritative
+information, only falling back on simulation if absolutely necessary.
```

Comment:
Personally I would advise more strongly against mocking, since the costs are borne by us (at least when the mocks appear in Google's monorepo), because any change to the actual build language risks causing each client of the the mocks to fail, and we are the ones that have to clean up the mess.

Reply:
Tweaked to:

> Any tooling that operates on BUILD and .bzl files must
carefully consider whether it is feasible to ask Bazel for the authoritative
information. The alternative, falling back on simulation, not only produces
less accurate results, but ties our hands as we try to improve Bazel.

Reply:
That's better.